Library/Homebrew/download_strategy.rb

```diff
@@ -407,24 +407,27 @@ def fetch(timeout: nil)
 
         ohai "Downloading #{url}"
 
-        use_cached_location = cached_location.exist?
-        use_cached_location = false if version.respond_to?(:latest?) && version.latest?
+        cached_location_valid = cached_location.exist? && !(version.respond_to?(:latest?) && version.latest?)
```

Comment:
I know this was pre-existing, but the use of `respond_to?` means this isn't typesafe, do you what the potential types are? (e.g. if `latest?` is only for the `Cask::DSL::Version` [case](https://github.com/Homebrew/brew/blob/62784545d0141483a40226574b87ab5e2cedb3bb/Library/Homebrew/cask/dsl/version.rb#L93), than I'd like us to use a static helper method in that class, use `to_s` directly, or some other typesafe approach.)

🤔 Maybe this will nerd snipe me into typing this file…

Reply:
'twas my plan all along, bwa ha ha